tempto-kafka/src/main/java/io/prestosql/tempto/fulfillment/table/kafka/KafkaTableManager.java

```diff
@@ -101,53 +100,36 @@ private void verifyTableExistsInPresto(String schema, String name)
 
     private void deleteTopic(String topic)
     {
-        withZookeeper(zkUtils -> {
-            if (AdminUtils.topicExists(zkUtils, topic)) {
-                AdminUtils.deleteTopic(zkUtils, topic);
-
-                for (int checkTry = 0; checkTry < 5; ++checkTry) {
-                    if (!AdminUtils.topicExists(zkUtils, topic)) {
-                        return;
-                    }
-                    try {
-                        Thread.sleep(1_000);
-                    }
-                    catch (InterruptedException e) {
-                        Thread.currentThread().interrupt();
-                        throw new RuntimeException("could not delete topic " + topic);
-                    }
-                }
-                throw new RuntimeException("could not delete topic " + topic);
+        AdminClient kafkaAdminClient = KafkaAdminClient.create(getKafkaProperties());
+
+        try {
+            ListTopicsResult topics = kafkaAdminClient.listTopics();
+            Set<String> names = topics.names().get();
+
+            if (names.contains(topic)) {
+                kafkaAdminClient.deleteTopics(ImmutableList.of(topic)).all().get();
             }
-        });
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Could not delete topic " + topic, e);
+        }
     }
 
     private void createTopic(String topic, int partitionsCount, int replicationLevel)
     {
-        withZookeeper(zkUtils -> {
-            Properties topicConfiguration = new Properties();
-            AdminUtils.createTopic(zkUtils, topic, partitionsCount, replicationLevel, topicConfiguration, RackAwareMode.Disabled$.MODULE$);
-        });
+        AdminClient kafkaAdminClient = KafkaAdminClient.create(getKafkaProperties());
+
+        try {
+            kafkaAdminClient.createTopics(ImmutableList.of(new NewTopic(topic, partitionsCount, toShortExact(replicationLevel)))).all().get();
```

Comment:
we want to also verify that replication is > 0.
Maybe just inline both checks (>0 and <MAX_SHORT) here via `verify`

Reply:
It's already checked in `KafkaTableDefinition`

`checkArgument(replicationLevel >= 1, "replicationLevel must be grater than or equal 1");`